src/lib/input/input.ts

```diff
@@ -282,6 +286,13 @@ export class MdInput implements ControlValueAccessor, AfterContentInit, OnChange
   ngAfterContentInit() {
     this._validateConstraints();
 
+    // Set the element type depending on normalized selector used(md-input / md-textarea)
+    if (this._elRef.nativeElement.tagName.toLowerCase() === 'md-input') {
+      this.elementType = 'input';
+    } else {
+      this.elementType = 'textarea';
+    }
```

Comment:
You should be able to do this in the constructor. I'd also make it a bit more concise with a ternary:
```ts
this.elementType = this._elementRef.nativeElement.nodeName.toLowerCase() === 'md-input' ?
    'input' : 
    'textarea';
```